packages/commonwealth/server/routes/ai/aiCompletionHandler.ts

```diff
@@ -16,9 +17,61 @@ export const aiCompletionHandler = async (req: Request, res: Response) => {
       useWebSearch = false,
     } = req.body as CompletionOptions;
 
-    // Validate inputs
-    if (!prompt) {
-      return res.status(400).json({ error: 'Prompt is required' });
+    let finalUserPrompt: string;
+    let finalSystemPrompt: string | undefined;
+
+    // Check if initialPrompt is an object with userPrompt and systemPrompt properties
+    if (
+      typeof initialPrompt === 'object' &&
+      initialPrompt !== null &&
+      typeof (initialPrompt as any).userPrompt === 'string'
+    ) {
+      finalUserPrompt = (initialPrompt as any).userPrompt;
+      // Use systemPrompt from the object if available, otherwise fallback to initialSystemPrompt (top-level)
+      finalSystemPrompt =
+        (initialPrompt as any).systemPrompt || initialSystemPrompt;
+      console.log('Interpreting structured prompt from initialPrompt field:');
+      console.log('  User Prompt (from object): ', finalUserPrompt);
+      if ((initialPrompt as any).systemPrompt) {
+        console.log(
+          '  System Prompt (from object): ',
+          (initialPrompt as any).systemPrompt,
+        );
+      }
+      if (initialSystemPrompt && !(initialPrompt as any).systemPrompt) {
+        console.log(
+          '  System Prompt (from top-level, as object did not have one): ',
+          initialSystemPrompt,
+        );
+      }
+    } else if (typeof initialPrompt === 'string') {
+      // This is the originally expected correct case
+      finalUserPrompt = initialPrompt;
+      finalSystemPrompt = initialSystemPrompt;
+      console.log('Interpreting flat prompt fields:');
+      console.log('  User Prompt: ', finalUserPrompt);
+      if (finalSystemPrompt) {
+        console.log('  System Prompt: ', finalSystemPrompt);
```

Comment:
a lot of console logs in this file, are they all necessary?

Reply:
will remove